CodeEdit/Features/QuickOpen/Extensions/Array+subscript_safe.swift

```diff
@@ -0,0 +1,15 @@
+//
+//  Array+subscript_safe.swift
+//  CodeEdit
+//
+//  Created by Paul Ebose on 2024/7/3.
+//
+
+import Foundation
+
+extension Array {
+    /// Returns nil if the index is out of bounds.
+    subscript(safe index: Index) -> Element? {
+        indices.contains(index) ? self[index] : nil
+    }
+}
```

Comment:
If I'm not mistaken, we already have this function available for you to use

Reply:
Interesting, it's in the GitClient.swift file. An extension on Collection.

Reply:
Swift’s array, set, and dictionary types are implemented as generic collections.
Could you please move the file to a more appropriate location? I think we have a folder designated for global extensions.